apis/beacon/deposit_snapshot.yaml

```diff
@@ -0,0 +1,34 @@
+  get:
+    operationId: getDepositSnapshot
+    summary: "Get Deposit Tree Snapshot"
+    description: |
+      Retrieve [EIP-4881](https://eips.ethereum.org/EIPS/eip-4881) Deposit Tree Snapshot.
+      Depending on `Accept` header it can be returned either as json or as bytes serialzed by SSZ
+    tags:
+      - Beacon
+    responses:
+      "200":
+        description: Success
+        content:
+          application/json:
+            schema:
+              type: object
+              title: GetDepositSnapshotResponse
+              properties:
+                data:
+                  $ref: '../../beacon-node-oapi.yaml#/components/schemas/DepositSnapshotResponse'
+          application/octet-stream:
+            schema:
+              description: "SSZ serialized block bytes. Use Accept header to choose this response type"
+      "404":
+        description: "No Finalized Snapshot Available"
+        content:
+          application/json:
+            schema:
+              allOf:
+                - $ref: "../../beacon-node-oapi.yaml#/components/schemas/ErrorMessage"
+                - example:
+                    code: 404
```

Comment:
we use 404 for when the "thing" is known not to exist (for example an empty slot) - a different code would probably be useful here for ephemeral error conditions (503/syncing for example)

Reply:
I do tend to agree 404 isn't ideal (a server with a route that is missing is also 404) - even if it was a 400, or potentially 503 might work... I'll see if i can get some more opinions on this one.